### ISSUE TITLE
multigpu training

### ISSUE THREAD

**Issue by thivux** [nguyenhoanganh2002/XTTSv2-Finetuning-for-New-Languages#11]
thanks a bunch for sharing this awesome work – it's super helpful! 😄

quick question: is it possible to use the training scripts for dvae and gpt with a multi-gpu setup?

keep up the great work, and thanks for being so responsive to everyone's questions! 🙏

**Comment by nguyenhoanganh2002** [comment]
> thanks a bunch for sharing this awesome work – it's super helpful! 😄
> 
> quick question: is it possible to use the training scripts for dvae and gpt with a multi-gpu setup?
> 
> keep up the great work, and thanks for being so responsive to everyone's questions! 🙏

Sorry for delayed reply. Currently, the training scripts are designed for single-GPU use. However, i'm considering implementing multi-GPU support in a future update.

**Comment by TugdualKerjan** [comment]
Using python -m trainer.distribute --gpus 0,1,2,... (whatever amount you have) --script train_xtts_gpt.py worked for me - beware that you have to deactivate the HFargparser in the train_xtts_gpt.py and instead pass the arguments as hardcoded vars in the script instead

**Comment by hellcatmon** [comment]
Hi @TugdualKerjan 

Have you made any adjustments to the data loader to enable it to run in parallel?

**Comment by hellcatmon** [comment]
Hi @anhnh2002 

Do you have any updates regarding the topic? Thanks

**Comment by TugdualKerjan** [comment]
I don't think so, are you having issues with the dataloader when using DDP ?

**Comment by hellcatmon** [comment]
It appears that the same dataset (a full copy) is being used for the second GPU. Is it possible to split a single set between GPUs instead?

**Comment by svsvec** [comment]
When do you plan to update multi-GPU support?

**Comment by cod3r0k** [comment]
Hi, can we run it in Multi-GPU mode? @TugdualKerjan @hellcatmon @thivux @svsvec @anhnh2002 Do you have any further updates?

**Comment by TugdualKerjan** [comment]
You can run it in multigpu, but again it's using DDP and this might not be optimized correctly. Look into the python trainer.distribute file

**Comment by hellcatmon** [comment]
I've slightly improved training performance on NVIDIA cards by switching from FP32 to TF32, which required only minor code adjustments. I also think it's feasible to switch to BF16, potentially increasing speed further, though a drop in accuracy might occur. If you're renting GPUs for training, the RTX 4090 currently provides the best performance-to-price efficiency, particularly after the switch to TF32. However, keep in mind that changes in pricing and demand could affect this assessment.

So it was enough to run in a single mode for our dataset.

For example, you could find GPU pricing there: [vast.ai](http://vast.ai)
